vendor/sigs.k8s.io/cloud-provider-azure/pkg/azureclients/fileclient/azure_fileclient.go

```diff
@@ -200,3 +200,74 @@ func (c *Client) GetServiceProperties(resourceGroupName, accountName string) (st
 func (c *Client) SetServiceProperties(resourceGroupName, accountName string, parameters storage.FileServiceProperties) (storage.FileServiceProperties, error) {
 	return c.fileServicesClient.SetServiceProperties(context.Background(), resourceGroupName, accountName, parameters)
 }
+
+// CreateSnapshot creates a snapshot
+func (c *Client) CreateSnapshot(ctx context.Context, resourceGroupName, accountName string, shareOptions *ShareOptions) (storage.FileShare, error) {
```

Comment:
it's ok for POC, while I think it's better reuse existing `CreateFileShare` func for taking snapshot

Reply:
`CreateFileShare` func doesn't return FileShare but we should use it to get tp time

Reply:
then change  `CreateFileShare` func a little,  return `storage.FileShare`  in that func.

Reply:
it's better reuse as possible as you can, that could also reduce the maintenance cost in long term

Reply:
ok, I will try it.